python/lsst/daf/butler/core/location.py

```diff
@@ -414,6 +432,19 @@ def __init__(self, datastoreRoot):
     def __str__(self):
         return f"{self.__class__.__name__}@{self._datastoreRootUri}"
 
+    @property
+    def bucketName(self):
+        """If Location is an S3 protocol, returns the bucket name."""
+        if self._datastoreRootUri.scheme == 's3':
+            return self._datastoreRootUri.netloc
+        else:
+            raise AttributeError(f'Path {self} is not a valid S3 protocol.')
+
+    @property
+    def datastoreRootName(self):
+        """Returns the name of the directory used as datastore's root."""
+        return os.path.basename(self._datastoreRootUri.path)
```

Comment:
Why does this exist? Firstly, it needs to understand that `path` might be `os.path` or `posixpath` depending on URI scheme. Secondly the name is confusing and the description because this is not the name of the directory used as the datastore's root, it's returning the final component of the path without any context as to where in the file system it really is.  Are you using this to work out some unique name for the datastore to use as a label? This seems wrong.

Reply:
This was a mistake. Leftovers I didn't catch previously. Removed.